### PR TITLE
extract google_credentials to `aws-google` gem

### PR DESCRIPTION
**Pull request by wjordan**
Extracts the code implementing Google Account integration with AWS into a separate [`aws-google`](https://github.com/code-dot-org/aws-google) gem.

The extracted gem also contains three improvements upon the current implementation in this repo:

1. Supports environments without browser; if browser is not detected, then the 'out of band' login flow will be triggered instead, which involves logging into a Google Account from an out-of-band browser, and copying/pasting a generated code into the terminal window where AWS access is requested.
2. Supports online-only authentication. In this mode (which can be enabled by passing the `online: true` option to the `Aws::Google` constructor), only a temporary Google access token will be returned by the refresh token (which is then used to create a temporary AWS session credentials), and no refresh token will be returned by the login flow. This way, no long-lived credentials will be stored on the filesystem, but the security/convenience tradeoff is that the browser-based login flow will need to be performed on a regular basis to refresh session credentials. For this reason I've left this disabled by default for now.
3. Support environments that don't have the AWS CLI installed. If the `aws` command is not detected, then credentials won't be written to the shared credentials file (which requires `aws configure set`), instead the credentials will be re-created every time the Ruby application starts.

### CODE REVIEW

File: bin/aws_access [hunk 1]
```diff
@@ -3,6 +3,16 @@
 require_relative '../deployment'
 require 'aws-sdk-sts'
 
+module Aws
+  class CredentialProviderChain
+    def current_provider
+      providers.find do |method_name, options|
+        send(method_name, options.merge(config: @config))&.set?
```
Comment (uponthesun) [inline bin/aws_access:10]: Everything makes sense to me except this - why does this behavior have to be overridden now?

**Comment by uponthesun** [comment]
Can one of the admins verify this patch?

File: bin/aws_access (hunk 1 above)
Comment (wjordan) [reply bin/aws_access:10]: This behavior-override was only moved, from its previous location in `cdo_google_credentials.rb` more narrowly into `bin/aws_access` (which is the only script that uses it).

Previously the `Aws::CredentialProviderChain` class was overridden for a second purpose, which was moved into the `aws-google` gem to [`credential_provider.rb`](https://github.com/code-dot-org/aws-google/blob/master/lib/aws/google/credential_provider.rb). So now that there's only one behavior-override left in this codebase, it made more sense to move it directly into the script.

**Comment by uponthesun** [comment]
test this please

**Comment by wjordan** [comment]
update: this is now working with our Docker container setup (no browser but `xdg-open` installed, which confused the launcher in the previous version). I plan to merge this once CI passes.

**Pull request merged**